src/services/panel.ts

```diff
@@ -0,0 +1,59 @@
+import { DataFrame } from '@grafana/data';
+import { SceneDataTransformer, SceneQueryRunner } from '@grafana/scenes';
+import { map, Observable } from 'rxjs';
+import { LokiQuery } from './query';
+import { explorationDS } from './variables';
+
+// TODO: `FieldConfigOverridesBuilder` is not exported, so it can not be used here.
+export function levelOverrides(overrides: any) {
+  overrides.matchFieldsWithName('info').overrideColor({
+    mode: 'fixed',
+    fixedColor: 'semi-dark-green',
+  });
+  overrides.matchFieldsWithName('debug').overrideColor({
+    mode: 'fixed',
+    fixedColor: 'semi-dark-blue',
+  });
+  overrides.matchFieldsWithName('error').overrideColor({
+    mode: 'fixed',
+    fixedColor: 'semi-dark-red',
+  });
+  overrides.matchFieldsWithName('warn').overrideColor({
+    mode: 'fixed',
+    fixedColor: 'semi-dark-orange',
+  });
+}
+
+export function sortLevelTransformation() {
+  return (source: Observable<DataFrame[]>) => {
+    return source.pipe(
+      map((data: DataFrame[]) => {
+        return data.sort((a, b) => {
+          const aName: string | undefined = a.fields[1].config.displayNameFromDS;
+          const aVal = aName?.includes('error') ? 4 : aName?.includes('warn') ? 3 : aName?.includes('info') ? 2 : 1;
+          const bName: string | undefined = b.fields[1].config.displayNameFromDS;
+          const bVal = bName?.includes('error') ? 4 : bName?.includes('warn') ? 3 : bName?.includes('info') ? 2 : 1;
+          return aVal - bVal;
+        });
+      })
+    );
+  };
+}
+
+export function getQueryRunner(query: LokiQuery) {
+  // if there's anything related to `level`, we want to sort the output equally
+  if (query.expr.includes('level')) {
```

Comment:
```suggestion
  if (query.expr.includes('LEVEL_VARIABLE_VALUE')) {
```

I also think we should make this check more reliable - because `detected_level` can be anywhere in query - even in log query. Maybe `(detected_level)` to ensure it is used in `sum by (...`?

Reply:
In fact the `level` was on purpose here, as all labels/fields graphs that are like a level should be ordered similar. See `detected_level` and `level` in 
![image](https://github.com/grafana/explore-logs/assets/8092184/d0b6f093-6ae7-47cc-bd72-19a41105270a)

Reply:
I just changed to use `legendFormat` in 15e9ed6a354108a234d84ecd209630a68c31dfd4. I think that should be reliant enough.

Reply:
Smart! I'd add a comment to provide context on why this is not `LEVEL_VARIABLE_VALUE` and `level`.